test/base.py

```diff
@@ -15,23 +15,52 @@
 
 import dataclasses
 import json
-from typing import Any, Callable, Optional
 import warnings
+from typing import Any, Callable, Optional
 
 import numpy as np
 from qiskit.test import QiskitTestCase
+from qiskit_experiments.calibration_management import Calibrations
+from qiskit_experiments.database_service.db_experiment_data import ExperimentStatus
 from qiskit_experiments.framework import (
     ExperimentDecoder,
     ExperimentEncoder,
+    ExperimentData,
     BaseExperiment,
     BaseAnalysis,
 )
-from qiskit_experiments.calibration_management import Calibrations
 
 
 class QiskitExperimentsTestCase(QiskitTestCase):
     """Qiskit Experiments specific extra functionality for test cases."""
 
+    def assertComplete(self, experiment_data: ExperimentData):
```

Comment:
To me this name is misleading, since the test blocks it isn't technically complete when you call this function, the function waits for the experiment to complete after calling. Also its a bit vague in general and should be something like `assertExperimentDone` since a cancelled or errored test is still "complete".